appcli/commands/main_cli.py

```diff
@@ -94,3 +111,121 @@ def orchestrator(ctx):
+        self,
+        cli_context: CliContext,
+        config_repo: ConfigurationGitRepository,
+        generated_config_repo: GeneratedConfigurationGitRepository,
+        force: bool = False,
+    ):
+        """Check the status of the configuration repositories and return any error messages.
+
+        Args:
+            cli_context (CliContext): the current cli context
+            config_repo (ConfigurationGitRepository): the configuration repository to check
+            generated_config_repo (GeneratedConfigurationGitRepository): the generated configuration repository to check
+            force (bool): whether to force (i.e. skip) errors. True to list errors as warnings, or false to exit and error. Defaults to False.
+        """
+
+        errors = []
+        # Check if the configuration directory contains unapplied changes
+        logger.debug("Checking if config repo is dirty")
+        if config_repo.is_dirty(untracked_files=True):
+            errors.append(
+                "Configuration contains un-applied changes. Call 'configure apply' to apply these changes."
+            )
+
+        # Check if the generated configuration repository has manual modifications to tracked files
+        logger.debug("Checking if generated config repo is dirty")
+        if generated_config_repo.is_dirty(untracked_files=False):
+            errors.append(
+                f"Generated configuration at [{generated_config_repo.repo_path}] has been manually modified."
+            )
+
+        # Check if the generated configuration is against current configuration commit #
+        logger.debug(
+            "Checking generated config metadata commit # matches config repo commit #"
+        )
+        metadata_file = get_metadata_file_directory(cli_context)
+        if not os.path.isfile(metadata_file):
+            errors.append(f"Could not find a metadata file at [{metadata_file}]")
+        else:
+            with open(metadata_file, "r") as f:
+                metadata = json.load(f)
+                logger.debug("Metadata from generated configuration: %s", metadata)
+
+            generated_conf_metadata_commit_hash = metadata["configure"]["apply"][
+                "commit_hash"
+            ]
+            configuration_commit_hash = config_repo.get_current_commit_hash()
+            if generated_conf_metadata_commit_hash != configuration_commit_hash:
+                errors.append(
+                    "Mismatched hashes between applied and current configuration. "
+                    + f"Configuration hash: [{configuration_commit_hash}], "
+                    + f"Applied hash: [{generated_conf_metadata_commit_hash}]"
+                )
+
+        if errors:
+            message = "\n".join(errors)
+            if not force:
+                error_and_exit(
+                    message
+                    + "\nOverride these warnings and continue to start by passing the '--force' flag."
+                )
+            logger.warn(
```

Comment:
Is warn too high? They did provide the --force flag after all.

Reply:
Fine for now